packages/embarkjs/src/utils.js

```diff
@@ -53,27 +57,90 @@ let Utils = {
 
       //toSend.estimateGas()
       //  .then(gasEstimated => {
-          //params.gas = gasEstimated + 1000;
-          return toSend.send(params, function(err, transactionHash) {
-            if (err) {
-              return callback(err);
-            }
-            hash = transactionHash;
-            hashCb(hash);
-          })
-          .on('receipt', function(receipt) {
-            if (!isContractDeploy || receipt.contractAddress) {
-              callback(null, receipt);
-            }
-          }).then(function(_contract) {
-            if (!hash) {
-              return; // Somehow we didn't get the receipt yet... Interval will catch it
-            }
-            web3.eth.getTransactionReceipt(hash, callback);
-          }).catch(callback);
-        //});
+      //params.gas = gasEstimated + 1000;
+      return toSend.send(params, function(err, transactionHash) {
+        if (err) {
+          return callback(err);
+        }
+        hash = transactionHash;
+        (hashCb || _hashCb)(hash);
+      })
+        .on('receipt', function(receipt) {
+          if (!isContractDeploy || receipt.contractAddress) {
+            callback(null, receipt);
+          }
+        }).then(function(_contract) {
+          if (!hash) {
+            return; // Somehow we didn't get the receipt yet... Interval will catch it
+          }
+          web3.eth.getTransactionReceipt(hash, callback);
+        }).catch(callback);
+      //});
     });
+
+    if (cb) { promise.then(_cb).catch(_cb); return; }
+    return promise;
   }
 };
 
 export default Utils;
+
+var getOwnPropertyDescriptors = Object.getOwnPropertyDescriptors ||
+    function getOwnPropertyDescriptors(obj) {
+      var keys = Object.keys(obj);
+      var descriptors = {};
+      for (var i = 0; i < keys.length; i++) {
+        descriptors[keys[i]] = Object.getOwnPropertyDescriptor(obj, keys[i]);
+      }
+      return descriptors;
+    };
+
+function callbackifyOnRejected(reason, cb) {
+  // `!reason` guard inspired by bluebird (Ref: https://goo.gl/t5IS6M).
+  // Because `null` is a special error value in callbacks which means "no error
+  // occurred", we error-wrap so the callback consumer can distinguish between
+  // "the promise rejected with null" or "the promise fulfilled with undefined".
+  if (!reason) {
+    var newReason = new Error('Promise was rejected with a falsy value');
+    newReason.reason = reason;
+    reason = newReason;
+  }
+  return cb(reason);
+}
+
+export function callbackify(original) {
+  if (typeof original !== 'function') {
+    throw new TypeError('The "original" argument must be of type Function');
+  }
+
+  // We DO NOT return the promise as it gives the user a false sense that
+  // the promise is actually somehow related to the callback's execution
+  // and that the callback throwing will reject the promise.
+  function callbackified() {
+    var args = [];
+    for (var i = 0; i < arguments.length; i++) {
```

Comment:
Can be replaced by 
```
args = Array.from(arguments).map(arg => arg);
```

Reply:
See: https://github.com/embark-framework/embark/pull/1323#discussion_r255625599.

Reply:
I can clean this up if it's deemed important, but really what I'd like is to figure out how to get webpack to use a newer version of the util shim, since then we could just remove this code.

Reply:
See https://github.com/embark-framework/embark/pull/1323#discussion_r255754542.